### DISCUSSION TITLE
OpenSSL 3.x.How to enable legacy algorithm support?

### DISCUSSION THREAD

**Discussion by wolf-hunter404** [openssl/openssl#24506] [Q&A]
The system is written in C++,the OpenSSL library is now upgraded to version 3.x.
In order to ensure that the project can be upgraded smoothly, functional testing is performed.
The parsing failed when calling the **PKCS12_parse** function. After debugging, it was found that some PKCS12 data used an old encryption algorithm, which is no longer supported by default. However, the data has been distributed and cannot be directly replaced. How should this be handled in the code?

**Comment by t8m** [comment]
1. you need to load the default and legacy providers
2. you need to verify that they loaded successfully - i.e. OSSL_PROVIDER_load() does not return NULL for them.

**Reply by wolf-hunter404** [reply]
OSSL_PROVIDER_load(NULL, "legacy");
OSSL_PROVIDER_load(NULL, "default");
After loading, the pointer is returned and there is no failure.
But the parsing still fails, the data can be confirmed to be correct.
